lib/internal/modules/esm/resolve.js

```diff
@@ -272,7 +168,7 @@ function fileExists(url) {
  * 4. TRY(pkg_url/index.js, pkg_url/index.json, pkg_url/index.node)
  * 5. NOT_FOUND
  * @param {URL} packageJSONUrl
- * @param {PackageConfig} packageConfig
+ * @param {import('internal/modules/esm/package_config.js').PackageConfig} packageConfig
```

Comment:
Optional suggestion: "import" the `PackageConfig` type with a typedef at the top of the file?